src/api/auth/src/hash.js

```diff
@@ -0,0 +1,20 @@
+// TODO: figure out how to share this...
+
+const crypto = require('crypto');
+
+/**
+ * Hash function used to generate our unique data ids.
+ * We use sha256 and encode in hex, so it's safe to use
+ * in URLs. For example:
+ *
+ *   6Xoj0UXOW3FNirlSYranli5gY6dDq60hs24EIAcHAEc=
+ *
+ * but truncate to only use the first 10 characters
```

Comment:
This is a good approach, iirc, it's also has better performance than using the entire digest size